src/app_service/service/stickers/async_tasks.nim

```diff
@@ -3,87 +3,25 @@ include ../../../app/core/tasks/common
-
-proc getPackData*(contract: ContractDto, getPackDataMethod: MethodDto, id: Stuint[256], running: var Atomic[bool]): StickerPackDto =
-  let secureSSLContext = newContext()
-  let client = newHttpClient(sslContext = secureSSLContext)
+proc getMarketStickerPacks*(running: var Atomic[bool], chainId: int): Table[string, StickerPackDto] =
+  result = initTable[string, StickerPackDto]()
   try:
-    let
-      contractMethod = getPackDataMethod
-      getPackData = GetPackData(packId: id)
-      payload = %* [{
-        "to": $contract.address,
-        "data": contractMethod.encodeAbi(getPackData)
-        }, "latest"]
-    let response = eth.doEthCall(payload)
-    if not response.error.isNil:
-      raise newException(RpcException, "Error getting sticker pack data: " & response.error.message)
-
-    let packData = decodeContractResponse[PackData](response.result.getStr)
-
-    if not running.load():
-      trace "Sticker pack task interrupted, exiting sticker pack loading"
-      return
-
-    # contract response includes a contenthash, which needs to be decoded to reveal
-    # an IPFS identifier. Once decoded, download the content from IPFS. This content
-    # is in EDN format, ie https://ipfs.infura.io/ipfs/QmWVVLwVKCwkVNjYJrRzQWREVvEk917PhbHYAUhA1gECTM
-    # and it also needs to be decoded in to a nim type
-    let contentHash = toHex(packData.contentHash)
-    let url = "https://ipfs.infura.io/ipfs/" & decodeContentHash(contentHash)
-    var ednMeta = client.getContent(url)
-
-    # decode the EDN content in to a StickerPackDto
-    result = edn_helper.ednDecode[StickerPackDto](ednMeta)
-    # EDN doesn't include a packId for each sticker, so add it here
-    result.stickers.apply(proc(sticker: var StickerDto) =
-      sticker.packId = truncate(id, int))
-    result.id = truncate(id, int)
-    result.price = packData.price
-  except Exception as e:
-    raise newException(RpcException, "Error getting sticker pack data: " & e.msg)
-  finally:
-    client.close()
-
-proc getAvailableStickerPacks*(
-    contract: ContractDto,
-    getPackCount: MethodDto,
-    getPackDataMethod: MethodDto,
-    running: var Atomic[bool]
-    ): Table[int, StickerPackDto] =
-
-  var availableStickerPacks = initTable[int, StickerPackDto]()
-  try:
-    let numPacksReponse = getPackCount(contract, getPackCount)
-
-    var numPacks = 0
-    if numPacksReponse.result.getStr != "0x":
-      numPacks = parseHexInt(numPacksReponse.result.getStr)
-
-    for i in 0..<numPacks:
-      if not running.load():
-        trace "Sticker pack task interrupted, exiting sticker pack loading"
-        break
-      try:
-        let stickerPack = getPackData(contract, getPackDataMethod, i.u256, running)
-        availableStickerPacks[stickerPack.id] = stickerPack
-      except:
-        continue
-    result = availableStickerPacks
+    let marketResponse = status_stickers.market(chainId)
+    if marketResponse.result.kind == JNull: return
```

Comment:
@richard-ramos in such cases it's better to check in a way... `if kind != expected_kind then return`
for this concrete case it could be:
```
if marketResponse.result.kind != JArray: 
    return
```